src/rerank/rerank.cpp

```diff
@@ -0,0 +1,137 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "rerank.hpp"
+
+#include <cmath>
+
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+#include "../logging.hpp"
+#include "../profiler.hpp"
+
+using namespace rapidjson;
+
+namespace ovms {
+
+absl::Status RerankHandler::parseRequest() {
+    // model: string
+    auto it = doc.FindMember("model");
+    if (it != doc.MemberEnd()) {
+        if (!it->value.IsString())
+            return absl::InvalidArgumentError("model accepts string values");
+        request.model = it->value.GetString();
+    }
+    // query: string
+    it = doc.FindMember("query");
+    if (it == doc.MemberEnd()) {
+        return absl::InvalidArgumentError("query field is missing in request");
+    }
+    if (!it->value.IsString()) {
+        return absl::InvalidArgumentError("query is not a string");
+    } else {
+        request.query = it->value.GetString();
+    }
+    // documents: list or map of strings
+    it = doc.FindMember("documents");
+    if (it == doc.MemberEnd()) {
+        return absl::InvalidArgumentError("documents field is missing in request");
+    }
+    if (!it->value.IsArray()) {
```

Comment:
Comment states:
```
// documents: list or map of strings
```

But only list is implemented. Will we need both? @michalkulakowski @dtrawins